allensdk/brain_observatory/behavior/behavior_project_cache.py

```diff
@@ -0,0 +1,311 @@
+        """
+        kwargs["manifest"] = kwargs.get("manifest",
+                                        "behavior_project_manifest.json")
+        kwargs["version"] = kwargs.get("version", self.MANIFEST_VERSION)
+
+        super().__init__(**kwargs)
+        self.fetch_api = fetch_api
+        self.fetch_tries = fetch_tries
+
+    @classmethod
+    def from_lims(cls, lims_kwargs=None, **kwargs):
+        lims_kwargs_ = lims_kwargs or {}
+        return cls(fetch_api=BehaviorProjectLimsApi.default(**lims_kwargs_),
+                   **kwargs)
+
+    def get_session_table(
+            self,
+            suppress: Optional[List[str]] = None) -> pd.DataFrame:
+        """
+        Return summary table of all ophys_session_ids in the database.
+        :param suppress: optional list of columns to drop from the resulting
+            dataframe.
+        :type suppress: list of str
+        :rtype: pd.DataFrame
+        """
+        def write_csv_(path, df):
+            """Format the array of experiment ids for saving"""
+            df_ = df.copy()
+            df_["ophys_experiment_id"] = df_["ophys_experiment_id"].apply(
+                lambda x: "|".join(map(str, x)))
+            df_.to_csv(path)
+
+        def read_csv_(path):
+            df = pd.read_csv(path, index_col="ophys_session_id")
+            df["ophys_experiment_id"] = df["ophys_experiment_id"].apply(
+                lambda x: np.fromstring(x, sep="|", dtype=int))
+            return df
+
+        sessions = self._get_session_summary(
+            self.fetch_api.get_session_table, self.OPHYS_SESSIONS_KEY,
+            write_csv_, read_csv_)
+        sessions = sessions.rename(columns={"genotype": "full_genotype"})
+        if suppress:
+            sessions.drop(columns=suppress, inplace=True, errors="ignore")
+        return sessions
+
+    def add_manifest_paths(self, manifest_builder):
+        manifest_builder = super().add_manifest_paths(manifest_builder)
+        for key, config in self.MANIFEST_CONFIG.items():
+            manifest_builder.add_path(key, **config)
+        return manifest_builder
+
+    def get_behavior_session_table(
+            self,
+            suppress: Optional[List[str]] = None) -> pd.DataFrame:
+        """
+        Return summary table of all behavior_session_ids in the database.
+        :param suppress: optional list of columns to drop from the resulting
+            dataframe.
+        :type suppress: list of str
+        :rtype: pd.DataFrame
+        """
+        read_csv_ = partial(read_csv, index_col="behavior_session_id")
+        sessions = self._get_session_summary(
+            self.fetch_api.get_behavior_only_session_table,
+            self.BEHAVIOR_SESSIONS_KEY, write_csv, read_csv_)
+        sessions = sessions.rename(columns={"genotype": "full_genotype"})
+        if suppress:
+            sessions.drop(columns=suppress, inplace=True, errors="ignore")
+        return sessions
+
+    def get_ophys_experiment_table(
```

Comment:
This seems like it ought to be supported by a separate query. ophys experiments have e.g. `calculated_depth`, which is not available on their session record.

Reply:
Thanks for the feedback. It is currently intended as a convenience method because `BehaviorOphysSession` is actually created with `ophys_experiment_id` rather than `ophys_session_id`. I would err on the side of not including this method, because I don't have requirements for it.

Reply:
Re: our conversation -- I'm inclined to add an additional keyword argument, `by`, where default is `by="ophys_session_id"`. If `by="ophys_experiment_id"`, then reindex so that we have one row per experiment id.